tools/src/main.rs

```diff
@@ -145,7 +158,8 @@ fn insert_residue_elements_and_commitments(
     Ok(reg.render_template(
         &verifier_contract_template,
         &json!({"residue_g2_elements": residue_g2_elements,
-                        "commitments": commitments}),
+                        "commitments": commitments,
+                        "vk_hash": vk_hash}),
```

Comment:
@koloz193 - but this vk_hash is not present in the verifier_contract_template.txt ?

Reply:
It’s one of the comments at the top (line 12)